chat-widget/src/components/livechatwidget/common/startChat.ts

```diff
@@ -75,11 +77,12 @@ const setPreChatAndInitiateChat = async (chatSDK: any, dispatch: Dispatch<ILiveC
 
     //Initiate start chat
     dispatch({ type: LiveChatWidgetActionType.SET_CONVERSATION_STATE, payload: ConversationState.Loading });
-    await initStartChat(chatSDK, props?.chatConfig, props?.getAuthToken, dispatch, setAdapter);
+    const optionalParams: StartChatOptionalParams = { isProactiveChat };
+    await initStartChat(chatSDK, props?.chatConfig, props?.getAuthToken, dispatch, setAdapter, optionalParams);
 };
 
 // eslint-disable-next-line @typescript-eslint/no-explicit-any
-const initStartChat = async (chatSDK: any, chatConfig: ChatConfig | undefined, getAuthToken: ((authClientFunction?: string) => Promise<string | null>) | undefined, dispatch: Dispatch<ILiveChatWidgetAction>, setAdapter: any, params?: any, persistedState?: any) => {
+const initStartChat = async (chatSDK: any, chatConfig: ChatConfig | undefined, getAuthToken: ((authClientFunction?: string) => Promise<string | null>) | undefined, dispatch: Dispatch<ILiveChatWidgetAction>, setAdapter: any, params?: StartChatOptionalParams, persistedState?: any) => {
```

Comment:
in my opinion, we need to start to explore the idea of using "Value objects" or wrappers to pass as input parameters,  this way to define the constructor is not the best, too many parameters.

Reply:
agreed. Will create a work item for eng. improvements